contrib/stack/topsStack/mergeBursts.py

```diff
@@ -283,51 +283,42 @@ def multilook(infile, outname=None, alks=5, rlks=15, multilook_tool="isce", no_d
     Take looks.
     '''
 
-    if multilook_tool=="gdal":
+    # default output filename
+    if outname is None:
+        spl = os.path.splitext(infile)
+        ext = '.{0}alks_{1}rlks'.format(alks, rlks)
+        outname = spl[0] + ext + spl[1]
 
+    if multilook_tool=="gdal":
         from osgeo import gdal
+        print(f"multilooking using gdal for {infile} ...")
 
-        print("multi looking using gdal ...")
-        if outname is None:
-            spl = os.path.splitext(infile)
-            ext = '.{0}alks_{1}rlks'.format(alks, rlks)
-            outname = spl[0] + ext + spl[1]
-        
-        print(infile)
         ds = gdal.Open(infile + ".vrt", gdal.GA_ReadOnly)
 
         xSize = ds.RasterXSize
         ySize = ds.RasterYSize
+        outXSize = int(xSize / int(rlks))
+        outYSize = int(ySize / int(alks))
+        srcXSize = outXSize * int(rlks)
+        srcYSize = outYSize * int(alks)
 
-        outXSize = xSize/int(rlks)
-        outYSize = ySize/int(alks)
-
-        if no_data:
-            gdalTranslateOpts = gdal.TranslateOptions(format="ENVI", width=outXSize, height=outYSize, noData=no_data)
-        else:
-            gdalTranslateOpts = gdal.TranslateOptions(format="ENVI", width=outXSize, height=outYSize)
+        options_str = f'-of ENVI -outsize {outXSize} {outYSize} -srcwin 0 0 {srcXSize} {srcYSize} '
+        options_str += f'-a_nodata {no_data}' if no_data else ''
+        gdal.Translate(outname, ds, options=options_str)
+        # repeat as a workaround solution to avoid the GDAL
```

Comment:
Can you elaborate a bit more here? So we have to repeat the same command to avoid an error?

Reply:
There is a gdal bug/issue, causing the multilooked geometry files not to be generated, as shown in #445. The more detailed cause within gdal is unknown. I actually tried this in python, the same lines of code work during debug mode, but failed in the command line while running the run_file or the SentinelWrapper.py command.

As suggested by @AlexApostolakis, calling this gdal.Translate functions twice, and the 2nd time call works, with the multilooked geometry files successfully generated.

Reply:
Try printing out `ds.GetFileList()` to see if some other files gets accidentally misinterpreted as belonging to the dataset. Maybe the file with the xml extension is

Reply:
Below is the print out of `ds.GetFileList()`. Another behavior is that if I commented out [the line to write the merged file to disk](https://github.com/isce-framework/isce2/blob/2bb3f0805dccfa7f81ef8a74b62ced32c34440e1/contrib/stack/topsStack/mergeBursts.py#L413), the later on gdal multilook issue is gone.

['/offset4motion/SaltonSeaSenDT173/merged/geom_reference/lat.rdr.full.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_01.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_02.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_03.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_04.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_05.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_06.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_07.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_08.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_09.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_10.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_11.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_12.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_13.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_14.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_15.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW1/lat_16.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_01.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_02.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_03.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_04.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_05.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_06.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_07.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_08.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_09.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_10.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_11.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_12.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_13.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_14.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_15.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW2/lat_16.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_01.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_02.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_03.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_04.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_05.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_06.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_07.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_08.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_09.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_10.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_11.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_12.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_13.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_14.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_15.rdr.vrt',
 '/offset4motion/SaltonSeaSenDT173/merged/geom_reference/../../geom_reference/IW3/lat_16.rdr.vrt']

Reply:
I suspect this is only happening for multiband files? 
https://github.com/isce-framework/isce2/blob/2bb3f0805dccfa7f81ef8a74b62ced32c34440e1/contrib/stack/topsStack/mergeBursts.py#L110 
suggests its using default scheme `BIP` from here: https://github.com/isce-framework/isce2/blob/2bb3f0805dccfa7f81ef8a74b62ced32c34440e1/components/isceobj/Image/Image.py#L86

whereas that GDAL line you mention explicitly sets it to BIL. Maybe the xml created by virtual VRT generation says its BIP and this not virtual thing is forcibly creating a BIL file - causing a mismatch at some point. 

Also, that `gdal_translate` call should become `gdal.Translate` like the others to avoid picking up different versions if possible.

Reply:
I just tried to replace the `gdal_translate` call with `gdal.Translate()` and changed its interleave to BIP. The error is still the same.

Reply:
Any particular type of file it fails on or it fails for everything?

Reply:
After discussing with @hfattahi offline, I confirmed the source of the issue is the existing *.hdr files. After removing them, the gdal.Translate call works again, as implemented in the commit below. Cheers!